File: peel/src/main/java/net/tjalp/peach/peel/config/Configurable.java
package net.tjalp.peach.peel.config;

/**
 * An interface that is applied to classes used
 * for [JsonConfig]
 */
public interface Configurable {

    /**
     * Called when the configuration
     * is loaded from the disk
     */
    default void onLoad() {}

    /**
     * Called when the configuration
     * is saved to the disk
     */
    default void onSave() {}
}

File: peel/src/main/kotlin/net/tjalp/peach/peel/config/Configurable.kt
package net.tjalp.peach.peel.config

/**
 * An interface that is applied to classes used
 * for [JsonConfig]
 */
interface Configurable {

    /**
    * Called when the configuration
    * is loaded from the disk
    */
    fun onLoad() {
        return
    }

    /**
     * Called when the configuration
     * is saved to the disk
     */
    fun onSave() {
        return
    }
}